internal/bsr/bsr.go

```diff
@@ -215,10 +216,22 @@ func OpenSession(ctx context.Context, sessionRecordingId string, f storage.FS, k
 		return nil, err
 	}
 
+	af, ok := SummaryAllocFuncs.Get(meta.Protocol, SessionContainer)
+	if !ok {
+		return nil, fmt.Errorf("%s: failed to get summary type", op)
+	}
+
+	summary := af(ctx)
+	if err := cc.decodeJsonFile(ctx, fmt.Sprintf(summaryFileNameTemplate, SessionContainer), summary); err != nil {
+		return nil, err
+	}
+	sessionSummary := summary.(*BaseSessionSummary)
```

Comment:
The Session struct expects `bsr.SessionSummary` instead of `bsr.Summary` so it complains when you try to pass `bsr.Summary` in. Unless you are saying all container structs should expect an interface of type `bsr.Summary`?

Reply:
Done